File: Py/abRepeat.py
t = int(input())

while t > 0:
    (n, k) = list(map(int, input().split(' ')))
    s = input()

    countsOfB = [0] * len(s)
    # print("initially counts of b is: ", countsOfB)

    if s[len(s) - 1] == 'b':
        countsOfB[len(s) - 1] = 1

    for index in range(len(s) - 2, -1, -1):
        # print("index = " + str(index) + " countsOfB[index] is: " + str(countsOfB[index]))
        countsOfB[index] = countsOfB[index + 1]
        if s[index] == 'b':
            countsOfB[index] += 1

    # print("now counts of b is(b on right of each char): ", countsOfB)

    totalBinString = countsOfB[0]

    totalAinString = 0
    for eachChar in s:
        if eachChar == 'a':
            totalAinString += 1

    # for index in range(0, len(countsOfB)):
    #     countsOfB[index] *= k

    # print("now counts of b is(*k): ", countsOfB)

    totalPairs = 0

    for index in range(0, len(s)):
        if s[index] == 'a':
            # print("inside if")
            totalPairs = totalPairs + (countsOfB[index] * k)

    extra = ((k * (k - 1)) // 2) * totalBinString * totalAinString
    # print("extra is: " + str(extra))

    print(totalPairs + extra)

    t -= 1

File: Cpp/abRepeat.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
  //   std::ios_base::sync_with_stdio(false);
  //   std::cin.tie(NULL);

  int t;
  std::cin >> t;

  while (t--) {
    long long int n, k;
    std::cin >> n >> k;

    std::string s;
    std::cin >> s;

    std::vector<long long int> countsOfB;
    countsOfB.resize(s.size(), 0);

    // std::cout << s;

    std::cout << " this is: " << s[s.size() - 1] << '\n';
    auto lastChar = s[s.size() - 1];
    if (lastChar == 'b') {
      std::cout << "inside if";
      countsOfB[s.size() - 1] = 1;
    }

    for (auto index = s.size() - 2; index >= 0; index--) {
      countsOfB[index] = countsOfB[index + 1];
      if (s[index] == 'b') {
        countsOfB[index]++;
      }
    }

    for (auto &eachCount : countsOfB) {
      eachCount *= k;
    }

    long long int totalPairs = 0;
    for (auto index = 0; index <= s.size(); index++) {
      if (s[index] == 'a') {
        totalPairs += (countsOfB[index]);
      }
    }

    std::cout << totalPairs << '\n';
  }

  return 0;
}